addons/website_slides_survey/controllers/slides.py

```diff
@@ -44,24 +45,47 @@ def _get_valid_slide_post_values(self):
     def _prepare_user_slides_profile(self, user):
         values = super(WebsiteSlides, self)._prepare_user_slides_profile(user)
         values.update({
-            'certificates': self._get_user_certificates(user),
+            'certificates': self._get_users_certificates(user)[user.id]
         })
         return values
 
     # All Users Page
     # ---------------------------------------------------
-    def _prepare_all_users_values(self, user):
-        result = super(WebsiteSlides, self)._prepare_all_users_values(user)
-        result.update({
-            'certification_count': len(self._get_user_certificates(user))
-        })
+    def _prepare_all_users_values(self, users):
+        result = super(WebsiteSlides, self)._prepare_all_users_values(users)
+        certificates_count_per_user = self._get_users_certificates_count(users)
+        for index, user in enumerate(users):
+            result[index].update({
+                'certification_count': certificates_count_per_user.get(user.partner_id.id, 0)
+            })
         return result
 
-    def _get_user_certificates(self, user):
+    def _get_users_certificates_count(self, users):
```

Comment:
Actually I still think it could be simplified. I think _get_users_certificates should return recordsets per user, and caller either takes the recordset or the length of it. When needing only count computation will be a bit less optimized but I am not sure it is heavy. It is a technical model with indexes, normally it should be quick. Moreover it is good to have this code now working in in batch, let us keep it readable :) .

Reply:
In short: have one method returning recordset, and take either recordset either its length according to what is necessary.